### PR TITLE
Feature/Hit the API

### PR DESCRIPTION
**Pull request by Benawi**
### General requirements
 - [x]  There are [no linter errors](https://github.com/microverseinc/linters-config)
 - [x]  The correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) is used
 - [x]  Readme file documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
 - [x] [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md) of javascript followed
### Project requirements
 - [x] The [Leaderboard API documentation](https://www.notion.so/microverse/Leaderboard-API-service-24c0c3c116974ac49488d4eb0267ade3) followed.
 - [x] A new game with the name of 'benawiGame' created by using the API.
 - [x] The ID of 'benawiGame game save that has  been returned by API.
 - [x] The "Refresh" button Implemented that will get all scores for a game created by you from the API (receiving data from the API and parsing the JSON).
 - [x] The form "Submit" button is implemented that will save a score for a game created (sending data to the API).
 - [x] Arrow functions used instead of the function keyword.
 - [x] async and await JavaScript features used to consume the API [L13-L22](https://github.com/Benawi/Microverse-Leaderboard/blob/3394e0e34b1f5ac793449a0927ee16e25bfe002b/src/modules/Store.js#L13-L22).
image

### CODE REVIEW

File: src/modules/Store.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    #privateSetData() {
+    this.data = fetch(this.scoresUrl)
+      .then((response) => response.json())
+      .then((json) => (json.result));
+  }
+
+    getData() {
+      this.#privateSetData();
+      return this.data;
+    }
```
Comment (VICTOR-CHUKWU) [inline src/modules/Store.js]: - [ ] According to the project requirement, all API calls should be done with `async await` method, so change all your `.then` methods to `async await`  :+1:

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Benawi  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Benawi** [comment]
### Changes ♻️
- [x] all .then methods changed to async await

**Comment by Benawi** [comment]
### Changes ♻️
- [x] All .then methods changed to async await 
- [x] Updated code sections:
       -  [L11-L34](https://github.com/Benawi/Microverse-Leaderboard/blob/aadb87d6f9bf64d5660de64ece2eecd54c972fde/src/modules/Store.js#L11-L34) and
       -  [LL23C1-L35C2](https://github.com/Benawi/Microverse-Leaderboard/blob/a159ce60e5db1d2e6c92d5732fee42c7a3ba5a28/src/modules/AddScore.js#LL23C1-L35C2)

**Review by AbeeraTahir — APPROVED** [review]
# Status: Approved ✔️✔️✔️  

Hi @Benawi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

approve

## Highlights
✔️ Linters are passing
✔️ Excellent implementation of changes suggested by the previous reviewer 👍

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**